pkg/generators/interface.go

```diff
@@ -18,6 +18,8 @@ type Generator interface {
 	// In case there is more then one generator the time will be the minimum of the times.
 	// In case NoRequeueAfter is empty, it will be ignored
 	GetRequeueAfter(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator) time.Duration
+
+	GetTemplate(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator) *argoprojiov1alpha1.ApplicationSetTemplate
```

Comment:
Suggest adding a  comment got `GetTemplate` here, something like `// GetTemplate returns the 'Template' field within the generator's CRD element`